frontend/packages/console-app/src/redux/actions/__tests__/cloud-shell-actions.spec.ts

```diff
@@ -0,0 +1,112 @@
+import { Dispatch } from 'react-redux';
+import {
+  setCloudShellExpanded,
+  setCloudShellActive,
+  Actions,
+  toggleCloudShellExpanded,
+} from '../cloud-shell-actions';
+import cloudShellConfirmationModal from '../../../components/cloud-shell/cloudShellConfirmationModal';
+
+const cloudShellConfirmationModalMock = cloudShellConfirmationModal as jest.Mock;
+
+jest.mock('../../../components/cloud-shell/cloudShellConfirmationModal', () => ({
+  default: jest.fn((action) => action()),
+}));
+
+describe('Cloud shell actions', () => {
+  beforeEach(() => {
+    cloudShellConfirmationModalMock.mockClear();
+  });
+
+  it('should create expand action', () => {
+    expect(setCloudShellExpanded(false)).toEqual({
+      type: Actions.SetCloudShellExpanded,
+      payload: {
+        isExpanded: false,
+      },
+    });
+    expect(setCloudShellExpanded(true)).toEqual({
+      type: Actions.SetCloudShellExpanded,
+      payload: {
+        isExpanded: true,
+      },
+    });
+  });
+
+  it('should create active action', () => {
+    expect(setCloudShellActive(false)).toEqual({
+      type: Actions.SetCloudShellActive,
+      payload: {
+        isActive: false,
+      },
+    });
+    expect(setCloudShellActive(true)).toEqual({
+      type: Actions.SetCloudShellActive,
+      payload: {
+        isActive: true,
+      },
+    });
+  });
```

Comment:
I don't think these are great tests 🤔 

For the most part you're testing `action` and static values. The _real_ test here is that you set `isActive` or `isExpanded` to the provided value. If we change the action in the future to have another property, this would fail and just need to be updated. If we changed the type, it's a constant to a constant change... so it would just need to be updated. If we changed the logic around setting `isExpanded` or `isActive` based on the provided value, that **does** have value as it changed the expected output against the signature of the method. ie, "I once provided `true`, but now you're consuming it as a number? A string? Inverting it? I didn't expect to that!". Whereas everything else mentioned is not impacting the signature => expected output of the action, more so just adding to it (around the initial expected output).

Reply:
Discussed with Andrew and we decided that using `expect.objectContaining` will suffice.